Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocDateTimeValueImplementation/Rendering/BocDateTimeValueRendererTest.cs

```diff
@@ -70,58 +70,57 @@ public class BocDateTimeValueRendererTest : RendererTestBase
+      _control.Setup (mock => mock.GetLabelIDs()).Returns (EnumerableUtility.Singleton (c_labelID));
+      _control.Setup (mock => mock.GetDateValueValidationErrors()).Returns (EnumerableUtility.Singleton (c_dateValidationErrors));
+      _control.Setup (mock => mock.GetTimeValueValidationErrors()).Returns (EnumerableUtility.Singleton (c_timeValidationErrors));
 
       _dateStyle = new SingleRowTextBoxStyle();
       _timeStyle = new SingleRowTextBoxStyle();
-      _control.Stub (stub => stub.DateTextBoxStyle).Return (_dateStyle);
-      _control.Stub (stub => stub.TimeTextBoxStyle).Return (_timeStyle);
+      _control.Setup (stub => stub.DateTextBoxStyle).Returns (_dateStyle);
+      _control.Setup (stub => stub.TimeTextBoxStyle).Returns (_timeStyle);
 
-      var pageStub = MockRepository.GenerateStub<IPage>();
-      pageStub.Stub (stub => stub.WrappedInstance).Return (new PageMock());
-      pageStub.Stub (stub => stub.ClientScript).Return (MockRepository.GenerateStub<IClientScriptManager>());
-      _control.Stub (stub => stub.Page).Return (pageStub);
+      var pageStub = new Mock<IPage>();
+      pageStub.Setup (stub => stub.WrappedInstance).Returns (new PageMock());
+      pageStub.Setup (stub => stub.ClientScript).Returns (new Mock<IClientScriptManager>().Object);
+      _control.Setup (stub => stub.Page).Returns (pageStub.Object);
 
-      var datePickerButton = MockRepository.GenerateStub<IDatePickerButton> ();
-      datePickerButton.Stub (stub => stub.EnableClientScript).Return (true);
-      datePickerButton.Stub (stub => stub.RenderControl (Html.Writer)).WhenCalled (invocation => Html.Writer.WriteLine ("DatePicker"));
-      _control.Stub (stub => stub.DatePickerButton).Return (datePickerButton);
+      var datePickerButton = new Mock<IDatePickerButton>();
+      datePickerButton.Setup (stub => stub.EnableClientScript).Returns (true);
+      datePickerButton.Setup (stub => stub.RenderControl (Html.Writer)).Callback ((HtmlTextWriter writer) => Html.Writer.WriteLine ("DatePicker"));
+      _control.Setup (stub => stub.DatePickerButton).Returns (datePickerButton.Object);
 
       StateBag stateBag = new StateBag ();
-      _control.Stub (stub => stub.Attributes).Return (new AttributeCollection (stateBag));
-      _control.Stub (stub => stub.Style).Return (_control.Attributes.CssStyle);
-      _control.Stub (stub => stub.DateTextBoxStyle).Return (new TextBoxStyle ());
-      _control.Stub (stub => stub.TimeTextBoxStyle).Return (new TextBoxStyle ());
-      _control.Stub (stub => stub.DateTimeTextBoxStyle).Return (new TextBoxStyle ());
-      _control.Stub (stub => stub.ControlStyle).Return (new Style (stateBag));
+      _control.Setup (stub => stub.Attributes).Returns (new AttributeCollection (stateBag));
+      _control.Setup (stub => stub.Style).Returns (_control.Object.Attributes.CssStyle);
+      _control.Setup (stub => stub.DateTimeTextBoxStyle).Returns (new TextBoxStyle ());
```

Comment:
What happened to
````
      _control.Stub (stub => stub.DateTextBoxStyle).Return (new TextBoxStyle ());
      _control.Stub (stub => stub.TimeTextBoxStyle).Return (new TextBoxStyle ());
```

Reply:
On line 84 the following expectations are set
```
_control.Stub (stub => stub.DateTextBoxStyle).Return (_dateStyle);
_control.Stub (stub => stub.TimeTextBoxStyle).Return (_timeStyle);
```
The expectations on line 100 will not override the old ones, therefore they can be deleted
```
_control.Stub (stub => stub.DateTextBoxStyle).Return (new TextBoxStyle ());
_control.Stub (stub => stub.TimeTextBoxStyle).Return (new TextBoxStyle ());
```
`_control` is a `Stub` and therefore does not override an already existing expectation. 
e.g.
```
var rhinoMocksMock = MockRepository.GenerateStub<Interface>();
rhinoMocksMock.Stub (_ => _.A).Return (2);
rhinoMocksMock.Stub (_ => _.A).Return (4);

Console.WriteLine (rhinoMocksMock.A); // 2
Console.WriteLine (rhinoMocksMock.A); // 2
```

In `Moq` every `Setup` can be overridden at any time.
```
var moqMock = new Mock<Interface>();
moqMock.Setup (_ => _.A).Returns (2);
moqMock.Setup (_ => _.A).Returns (4);

Console.WriteLine (moqMock.Object.A); // 4
Console.WriteLine (moqMock.Object.A); // 4
```

Maybe I'm missing something, but the deleted setup is not needed in the `Moq` version of the test

Reply:
Ah, thanks. the downside of looking at DIFFs.